### ISSUE TITLE
Bluebird version conflict

### ISSUE THREAD

**Issue by schmod** [RobinBuschmann/sequelize-typescript#165]
There appears to be a minor version incompatibility between the version of the Bluebird typings required by this project(3.5.4), and the latest version of those typings (3.5.15).  

 It may make sense to alter this project's `package.json` to play nicely with newer semver-minor releases of the Sequelize and Bluebird typings.  

`@types/sequelize` is _super-generous_, and actually just specifies `@types/bluebird: "*"` as a dependency). 

```
src/file.ts.ts(57,3): error TS90010: Type 'Bluebird<number>' is not assignable to type 'Bluebird<number>'. Two different types with this name exist, but they are unrelated.
  Types of property 'then' are incompatible.
    Type '{ <U1, U2>(onFulfill: (value: number) => U1 | Thenable<U1>, onReject: (error: any) => U2 | Thenab...' is not assignable to type '{ <U>(onFulfill?: ((value: number) => U | PromiseLike<U>) | undefined, onReject?: ((error: any) =...'.
      Type 'Bluebird<any>' is not assignable to type 'Bluebird<any>'. Two different types with this name exist, but they are unrelated.
        Types of property 'catch' are incompatible.
          Type '{ (onReject?: ((error: any) => any) | undefined): Bluebird<any>; <U>(onReject?: ((error: any) => ...' is not assignable to type '{ (onReject?: ((error: any) => any) | undefined): Bluebird<any>; <U>(onReject?: ((error: any) => ...'. Two different types with this name exist, but they are unrelated.
            Types of parameters 'onReject' and 'filter1' are incompatible.
              Type 'new (...args: any[]) => any' is not assignable to type '((error: any) => any) | undefined'.
                Type 'new (...args: any[]) => any' is not assignable to type '(error: any) => any'.
                  Type 'new (...args: any[]) => any' provides no match for the signature '(error: any): any'.
```

**Comment by Axure** [comment]
It's interesting to see that there's not an end to this issue...

**Comment by RobinBuschmann** [comment]
@Axure Can your describe your issue a little bit more - best in a new issue? The currently used `@types/sequelize` version defines its `@types/bluebird` dependency with `*`. Which means it doesn't require a specific version. So there can't be a conflict with the typings required by `@types/sequelize` anymore.

**Comment by spearmootz** [comment]
i have this same issue.

i have it on model methods

``` saveDate(date: moment.Moment) {
    this.data = date.format(momentFormat);
    return this.save();
  }```

if i do this then it doesnt like that it doesnt have typing. but if i put it returns Bluebird<this> then i also get the error above.

**Comment by spearmootz** [comment]
actually, i get this. 

Type 'Bluebird<this>' is not assignable to type 'Bluebird<this>'. Two different types with this name exist, but they are unrelated.
  Types of property 'nodeify' are incompatible.
    Type '{ (callback: (err: any, value?: this | undefined) => void, options?: SpreadOption | undefined): this; (...sink: any[]): this; }' is not assignable to type '{ (callback: (err: any, value?: this | undefined) => void, options?: SpreadOption | undefined): Bluebird<this>; (...sink: any[]): Bluebird<this>; }'.

**Comment by RobinBuschmann** [comment]
Did you installed `@types/bluebird` additionally? If so, for now you have to make sure, you've installed the same version as required by `sequelize-typescript`

Hmmm, probably we should go with the solution schmod initially suggested 🤔

**Comment by s00500** [comment]
I think I have a similar problem

```  
get(userId: number): Promise<User> {
    L.info(`fetch user with id ${userId}`);
    return (User.findById(userId) as unknown) as Promise<User>;
    //return User.findById(userId);
  }
```

Whit the weird cast everything works, without it I get:

```
server/api/services/user.service.ts(9,5): error TS2719: Type 'Bluebird<import("/Users/LB/Desktop/z_Projects\uF8FF\uF8FF\uF8FF/stoctopus/stoctopusServer/models/user").default>' is not assignable to type 'Bluebird<import("/Users/LB/Desktop/z_Projects\uF8FF\uF8FF\uF8FF/stoctopus/stoctopusServer/models/user").default>'. Two different types with this name exist,but they are unrelated.
```

**Comment by RobinBuschmann** [comment]
@s00500 Are you using the same @types/bluebird version sequelize-typescript requires?

**Comment by s00500** [comment]
@RobinBushmann i have 3.5.18 installed instead of the newer 3.5.24, looks correct in the package-lock file

**Comment by RobinBuschmann** [comment]
@s00500 probably another package that installed a different version?

Hmmm, I think we should loosen the semver range on @types/bluebird as schmod initially intended.

**Comment by s00500** [comment]
@RobinBuschmann no, searched my package-lock and did not find another version of the types

I am now wrapping the sequelize call into a new promise each time to avoid the ugly cast

**Comment by RobinBuschmann** [comment]
Solved on branch `1.0.0`

**Comment by nadrane** [comment]
hmm should this issue be resolved in `1.0.0-beta.3` and `1.0.0-alpha.9`

I am getting the exact same Bluebird conflict in both. Is there something I might be missing?

**Comment by RobinBuschmann** [comment]
@nadrane Hmm, that’s strange since sequelize-typescript doesn’t require a specific version anymore and also defines @types/bluebird as a peer dependency. Can you provide an example repo? I also think it might make sense to open a new issue for that. Thank you!
